### ISSUE TITLE
Feature Request - Use Android Activity Recognition API

### ISSUE THREAD

**Issue by dough29** [tananaev/traccar-client-android#293]
Hello,

I actually use a Traccar server with my home automation system. Cars have trackers (TK102) and Android phones use Macrodroid APP to share location with OsmAnd protocol.

I use macro with Android Activity Recognition API to share location at different frequencies :
- still : send location every 1 hour
- walking : every 30 minutes
- running : every 15 minutes
- biking : every 1 minutes
- in car on battery : every 2min30
- in car and charging : every 30 secondes

I think this can be a cool feature to add to Android (and iOS) Traccar Client app, what do you think ?

**Comment by tananaev** [comment]
It's part of Google APIs, so it's not part of open source.

**Comment by dough29** [comment]
Hmm what do you mean ?

That API just tells apps that subscribed to "Activity Recognition" whether user is still, walking, running, driving, ...

So if implemented the app can do actions like sharing location at desired frequency for each activity.

**Comment by tananaev** [comment]
Those APIs are proprietary and not available on phones without Google services.

**Comment by dough29** [comment]
Ok, so not available for custom devices but what about giving that option to people using Traccar Client on their daily phone ?

**Comment by tananaev** [comment]
What do you mean custom devices? There are plenty of devices without Google services.

**Comment by dough29** [comment]
I was thinking some people have old phones with nothing installed else the Traccar Client to have a cheap tracker device ?

Most of Android phones have Google services or am I mistaking ?

I didn't develop on Android for many years now so I think I'm outdated on this subject ^^

**Comment by tananaev** [comment]
It's more about the principle. I don't want open source project to depend on proprietary Google technology.

**Comment by dough29** [comment]
Yes I understand, by the way the user can choose to use or not the feature.

**Comment by OverkillTASF** [comment]
Tananaev, thanks for sticking to your principles on this one. I am one of those Traccar users who doesn't want any Google "services" on my phone.
